src/ecalc/libraries/libecalc/common/tests/dto/test_fuel_consumer.py

```diff
@@ -21,23 +103,83 @@ def test_missing_fuel(self):
                         energy_usage_type=EnergyUsageType.FUEL,
                     )
                 },
-                regularity={datetime(2000, 1, 1): Expression.setup_from_expression(1)},
+                regularity=regularity,
                 user_defined_category="category",
             )
         assert "Missing fuel for fuel consumer 'test'" in str(exc_info.value)
 
+    def test_duplicate_fuel_names(self):
+        """
+        TEST SCOPE: Check that duplicate fuel type names are not allowed.
+
+        Duplicate fuel type names should not be allowed across installations.
+        Duplicate names may lead to debug problems and overriding of previous
+        values without user noticing. This test checks that different fuels cannot
+        have same name.
+        """
+        fuel_consumer1 = get_fuel_consumer(
+            consumer_name="flare",
+            fuel_type=get_fuel("fuel1", price=100, emission_name="co2"),
+            category={datetime(2000, 1, 1): dto.base.ConsumerUserDefinedCategoryType.FLARE},
+        )
+
+        fuel_consumer2 = get_fuel_consumer(
+            consumer_name="boiler",
+            fuel_type=get_fuel("fuel1", price=1000, emission_name="ch4"),
+            category={datetime(2000, 1, 1): dto.base.ConsumerUserDefinedCategoryType.BOILER},
+        )
+
+        installation1 = get_installation("INST1", fuel_consumer1)
+        installation2 = get_installation("INST2", fuel_consumer2)
+
         with pytest.raises(ValidationError) as exc_info:
-            dto.FuelConsumer(
-                name="test",
-                fuel={},
-                component_type=ComponentType.GENERIC,
-                energy_usage_model={
-                    datetime(2000, 1, 1): dto.DirectConsumerFunction(
-                        fuel_rate=Expression.setup_from_expression(1),
-                        energy_usage_type=EnergyUsageType.FUEL,
-                    )
-                },
-                regularity={datetime(2000, 1, 1): Expression.setup_from_expression(1)},
-                user_defined_category="category",
+            dto.Asset(
+                name="multiple_installations_asset",
+                installations=[
+                    installation1,
+                    installation2,
+                ],
             )
-        assert "Missing fuel for fuel consumer 'test'" in str(exc_info.value)
+
+        assert "Duplicated names are: fuel1" in str(exc_info.value)
+
+    def test_same_fuel(self):
+        """
+        TEST SCOPE: Check that validation of duplicate fuel type names do not reject
+        when same fuel is used across installations.
+
+        Even though duplicate fuel type names are not allowed across installations,
+        it should be possible to re-use the same fuel. This test verifies that this still
+        works.
+        """
+
+        fuel_consumer1 = get_fuel_consumer(
```

Comment:
I think that this example here is somehow unproblematic. This can be a problem that the user names a fuel with the same name twice, but they are both accessible. The main problem here I guess is that it can be confusing wrt reporting and aggregating "emissions by fuel_name", then that would cause a problem.